src/platform_linux/shmem.c

```diff
@@ -198,13 +309,13 @@ platform_shm_hip(platform_heap_id hid)
 static inline void
 shm_lock_mem_frags(shmem_info *shminfo)
 {
-   platform_spin_lock(&shminfo->shm_mem_frags_lock);
+   platform_mutex_lock(&shminfo->shm_mem_frags_mutex);
```

Comment:
Better solution.... Otherwise, for some cases of new `large_inserts_stress_test`, we were simply burning up 100% CPU. 

In all workloads, pthread-semaphore is less than 5%, sometimes even 1-2% as seen in `perf top`.